posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,553 @@
+        Args:
+            train_path (str) : path/name of '.h5' file for training data.
+            valid_path (str) : path/name of '.h5' file for testing/validation data.
+            profile_names (array-like) : list of profile quantities to be extracted.
+        """
+        self.names = profile_names
+        
+        try:
+            print(valid.final_values["S1_state"][0])
+        except:
+            print("grid does not have S1_state information")
+        
+        # extract testing data
+        print("extracting testing data")
+        valid = PSyGrid(valid_path) # load PSyGrid
+        self.valid_scalars = pd.DataFrame()
+        self.valid_profiles = []
+        testing_failed = []
+
+        for i in range(len(valid)):
+            try:
+                scalars,profiles = self.scrape(valid,i)
+                self.valid_scalars = self.valid_scalars.append(scalars,ignore_index=True)
+                self.valid_profiles.append(profiles)
+            except:
+                testing_failed.append(i)
+                pass
+
+        # extract training data
+        print("extracting training data")
+        train = PSyGrid(train_path) # load PSyGrid
+        self.scalars = pd.DataFrame()
+        self.profiles = []
+        training_failed = []
+        
+        for i in range(len(train)):
+            try:
+                scalars,profiles = self.scrape(train,i)
+                self.scalars = self.scalars.append(scalars,ignore_index=True)
+                self.profiles.append(profiles)
+            except:
+                training_failed.append(i)
+                pass
+
+
+    def scrape(self,grid,ind):
+        """Extracts profile data from one MESA run.
+        Args:
+            grid (obj) : PSyGrid object.
+            ind (int) : index of run to be scraped.
+        Returns:
+            scalars (array-like) : dictionary containing initial m1, m2, p and final s1_state, final_mass.
+            profiles (array-like) : all N specified profiles, shape (N,200).
+        """
+        # open individual run as a DataFrame
+        df = pd.DataFrame(grid[ind]['final_profile1'])
+        df = df.sort_values(by="mass")
+        df = df.reset_index(drop=True)
+
+        # grab input values, final star 1 state, final star 1 mass
+        m1 = grid.initial_values["star_1_mass"][ind]
+        m2 = grid.initial_values["star_2_mass"][ind]
+        p = grid.initial_values["period_days"][ind]
+        state = grid.final_values["S1_state"][ind]  
+        final_mass = grid.final_values["star_1_mass"][ind]
+        scalars = {"m1":m1,
+                   "m2":m2,
+                   "p":p,
+                   "s1_state":state,
+                   "final_mass":final_mass}
+
+        # grab output vectors, interpolate to normalize
+        profiles=np.zeros([len(self.names),200])
+        for i,prof in enumerate(self.names):
+            f = interp1d(df['mass']/final_mass,df[prof],fill_value="extrapolate")
```

Comment:
You may want to check first, that the `df` contains the column `prof` and add a catch if it is not there.